item-impl/src/main/java/com/example/auction/item/impl/ItemServiceImpl.java

```diff
@@ -43,25 +44,32 @@ public ItemServiceImpl(PersistentEntityRegistry registry, BiddingService bidding
 
         registry.register(PItemEntity.class);
 
-        biddingService.bidEvents().subscribe().atLeastOnce(Flow.<BidEvent>create().mapAsync(1, event -> {
-            if (event instanceof BidEvent.BidPlaced) {
-                BidEvent.BidPlaced bidPlaced = (BidEvent.BidPlaced) event;
-                return entityRef(bidPlaced.getItemId())
-                        .ask(new PItemCommand.UpdatePrice(bidPlaced.getBid().getPrice()));
-            } else if (event instanceof BidEvent.BiddingFinished) {
-                BidEvent.BiddingFinished biddingFinished = (BidEvent.BiddingFinished) event;
-                PItemCommand.FinishAuction finishAuction = new PItemCommand.FinishAuction(
-                        biddingFinished.getWinningBid().map(Bid::getBidder),
-                        biddingFinished.getWinningBid().map(Bid::getPrice).orElse(0));
-                return entityRef(biddingFinished.getItemId()).ask(finishAuction);
-            } else {
-                // Ignore.
-                return CompletableFuture.completedFuture(Done.getInstance());
-            }
-        }));
+        biddingService.bidEvents().subscribe().atLeastOnce(Flow.<BidEvent>create().mapAsync(1, this::handleBidEvent));
     }
 
-    
+  /**
+   *  TODO: doc this.
+   * @param event
+   * @return
+   */
+    private CompletionStage<Done> handleBidEvent(BidEvent event) {
```

Comment:
Good idea extracting this, and I might go one step further and extract each event handler to a separate method as well.